File: DataDistributionManagerNet/Interop/UnderlyingEvent.cs
/*
*  Copyright 2021 MASES s.r.l.
*  
*  Licensed under the Apache License, Version 2.0 (the "License");
*  you may not use this file except in compliance with the License.
*  You may obtain a copy of the License at
*  
*  http://www.apache.org/licenses/LICENSE-2.0
*  
*  Unless required by applicable law or agreed to in writing, software
*  distributed under the License is distributed on an "AS IS" BASIS,
*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
*  See the License for the specific language governing permissions and
*  limitations under the License.
*  
*  Refer to LICENSE for more information.
*/

using System;
using System.Runtime.InteropServices;

namespace MASES.DataDistributionManager.Bindings.Interop
{
    /// <summary>
    /// UnderlyingEvent struct representing the information from low system.
    /// </summary>
    [StructLayout(LayoutKind.Sequential)]
    internal struct UnderlyingEvent
    {
        [MarshalAs(UnmanagedType.LPStr)]
        public string ChannelName;
        public DDM_UNDERLYING_ERROR_CONDITION Condition;
        [MarshalAs(UnmanagedType.U1)]
        public bool IsDataAvailable;
        public IntPtr Key;
        public IntPtr KeyLen;
        public IntPtr Buffer;
        public IntPtr BufferLength;
        [MarshalAs(UnmanagedType.I4)]
        public int NativeCode;
        [MarshalAs(UnmanagedType.LPStr)]
        public string SubSystemReason;
    }
}

File: src/DataDistributionManagerJava/src/org/mases/datadistributionmanager/UnderlyingEvent.java
/*
*  Copyright 2021 MASES s.r.l.
*
*  Licensed under the Apache License, Version 2.0 (the "License");
*  you may not use this file except in compliance with the License.
*  You may obtain a copy of the License at
*
*  http://www.apache.org/licenses/LICENSE-2.0
*
*  Unless required by applicable law or agreed to in writing, software
*  distributed under the License is distributed on an "AS IS" BASIS,
*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
*  See the License for the specific language governing permissions and
*  limitations under the License.
*
*  Refer to LICENSE for more information.
*/

package org.mases.datadistributionmanager;

/**
 * UnderlyingEvent struct representing the information from low system.
 */
public class UnderlyingEvent {
    UnderlyingEvent(String channelName, OPERATION_RESULT condition, boolean isDataAvailable, String key,
            byte[] buffer, int nativeCode, String subSystemReason) {
        this.ChannelName = channelName;
        this.Condition = condition;
        this.IsDataAvailable = isDataAvailable;
        this.Key = key;
        this.Buffer = buffer;
        this.NativeCode = nativeCode;
        this.SubSystemReason = subSystemReason;
    }

    final String ChannelName;
    final OPERATION_RESULT Condition;
    final boolean IsDataAvailable;
    final String Key;
    final byte[] Buffer;
    final int NativeCode;
    final String SubSystemReason;
}